### PR TITLE
Grafana-UI: Export Card container styles

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
It's impossible to cover all use cases with one `Card` component, so an alternative is to export the container styles, so they could be applied to other list components and make at least the basic styling consistent.

### CODE REVIEW

**Comment by torkelo** [comment]
I am not sure I like this, what is it you want t reuse? the Card container color & hover?

**Comment by torkelo** [comment]
Maybe we can have a CardContainer component? exporting styles like this feel very brittle

**Comment by Clarity-89** [comment]
> I am not sure I like this, what is it you want t reuse? the Card container color & hover?

I have an item list with content layout that's quite different from what's available with Card and I'd like the container styles to be the same. I saw a similar use case in other places as well. 

> Maybe we can have a CardContainer component? exporting styles like this feel very brittle

We have a `CardContainer` component, but the styles are applied in the `Card` itself. Not sure how exporting the component itself is that much different that exporting the styles? As an alternative we could export `CardContainer` and `CardInner` components to allow users create `Card` components with custom layout.

File: packages/grafana-ui/src/components/Card/CardContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    <Tooltip placement="top" content={tooltip} theme="info">
+      <div {...props} className={cx(container, className)}>
+        {children}
+      </div>
+    </Tooltip>
+  ) : (
+    <div {...props} className={cx(container, className)}>
+      {children}
+    </div>
+  );
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: feels strange that the whole card has a toolttip?

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: That's according to the designs, there are cases when we need to display tooltips on card hover, also it only applies when the card has a `tooltip` prop.

Comment (torkelo) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: ok, feels very strange to have a tooltip on something so large

Comment (torkelo) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: Where is this tooltip prop used?

Comment (torkelo) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: Cannot find this used anywhere, can we remove it?

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: Card can also be a list component with just a title and description. In that case having a tooltip makes sense there. It's just our Card component is trying to handle too many use cases imo.

Comment (torkelo) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: But if that comes up the user can just wrap it in a Tooltip component, just a big object like wide or high card a tooltip seems like a bad idea 90% of the time

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Card/CardContainer.tsx:35]: Just realised myself that users can just wrap the card in tooltip if they need it, good point, I'll remove it.

File: packages/grafana-ui/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -100,66 +63,41 @@ export const Card: CardInterface = ({
...
-    ...(!disableHover && {
-      '&:hover': {
-        background: theme.colors.emphasize(theme.colors.background.secondary, 0.03),
-        cursor: 'pointer',
-        zIndex: 1,
-      },
-      '&:focus': styleMixins.getFocusStyles(theme),
-    }),
-  });
-});
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Card/Card.tsx:161]: An alternative idea would be to make this into a mixin that returns a CSSObject that we can reuse in other Card like elements, a CSSObject feels easier and cleaner to re-use & extend compared to a css class, What elements is it you want to share for your new card? the background, hover & focus state? the inner link?

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Card/Card.tsx:161]: It's for report list view, I think `CadContainer` works pretty good, it also abstracts the logic that inner content can be a link or a div. 
Screenshot 2021-04-28 at 9 43 02

Comment (torkelo) [reply packages/grafana-ui/src/components/Card/Card.tsx:161]: looks more like table :)

**Review by torkelo — APPROVED** [review]

File: packages/grafana-ui/src/components/Card/Card.tsx (hunk 2 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Card/Card.tsx:161]: yeah, it used to be a table, but I thought it's easier to style it if it's just a Card item.

**Pull request merged**